spec/fixtures/custom_nginx.template

```diff
@@ -188,31 +199,53 @@ http {
             proxy_pass         $upstream_scheme://kong_upstream$upstream_uri;
         }
 
-        location = /kong_error_handler {
+        location @grpc {
             internal;
-            uninitialized_variable_warn off;
-
-            rewrite_by_lua_block {;}
+            set $kong_proxy_mode  'grpc';
 
-            access_by_lua_block {;}
+            grpc_set_header    TE                $upstream_te;
+            grpc_set_header    Host              $upstream_host;
+            grpc_set_header    X-Forwarded-For   $upstream_x_forwarded_for;
+            grpc_set_header    X-Forwarded-Proto $upstream_x_forwarded_proto;
+            grpc_set_header    X-Forwarded-Host  $upstream_x_forwarded_host;
+            grpc_set_header    X-Forwarded-Port  $upstream_x_forwarded_port;
+            grpc_set_header    X-Real-IP         $remote_addr;
+            grpc_pass_header   Server;
+            grpc_pass_header   Date;
+            grpc_ssl_name      $upstream_host;
 
-            content_by_lua_block {
-                Kong.handle_error()
-            }
+            grpc_pass grpc://kong_upstream;
         }
 
-        location @grpc {
+        location @grpcs {
             internal;
-
             set $kong_proxy_mode  'grpc';
-            grpc_pass grpc://kong_upstream;
+
+            grpc_set_header    TE                $upstream_te;
```

Comment:
Still unsure we should add this; [according to the spec](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests), it'll be set to `trailers`, so might be better to just leave it up to the grpc module.

Reply:
@gszr we have special treatment for `trailers` already:
https://github.com/Kong/kong/blob/master/kong/runloop/handler.lua#L1346-L1352